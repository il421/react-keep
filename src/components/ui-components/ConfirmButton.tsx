import React from "react";
import { BeatLoader } from "react-spinners";
import { FlexBox } from "./FlexBox";
import "../../styles/ui-components/_login-button.scss";
import { Colors, JustifyContent } from "../../common";

export interface ConfirmButtonProps
  extends Partial<Pick<HTMLButtonElement, "disabled" | "className" | "id">> {
  text: string;
  loading?: boolean;
  type?: "submit" | "button";
  onClick?: (evt?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  wrapperClassName?: string;
}

export class ConfirmButton extends React.PureComponent<ConfirmButtonProps> {
  render() {
    const {
      type = "submit",
      loading,
      className = "login-button",
      text,
      disabled = false,
      onClick,
      id,
      wrapperClassName,
    } = this.props;
    return (
      <FlexBox
        className={wrapperClassName}
        justifyContent={JustifyContent.center}
      >
        {!loading ? (
          <button
            id={id}
            className={className}
            type={type}
            disabled={disabled}
            onClick={onClick}
          >
            {text}
          </button>
        ) : (
          <BeatLoader
            size={10}
            color={Colors.fresh}
            css={
              "height: 42px; margin-bottom: 12px; display: flex; align-items: center"
            }
          />
        )}
      </FlexBox>
    );
  }
}
