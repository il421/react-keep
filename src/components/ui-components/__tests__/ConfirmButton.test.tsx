import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { ConfirmButton, ConfirmButtonProps } from "../ConfirmButton";

let props: ConfirmButtonProps, wrapper: ReactWrapper<ConfirmButtonProps, {}>;

beforeEach(() => {
  props = {
    text: "text",
    onCLick: jest.fn(),
  };
});

test("should render ConfirmButton", () => {
  wrapper = mount<ConfirmButtonProps>(<ConfirmButton {...props} />);
  expect(wrapper.debug()).toMatchSnapshot();
});

test("should render ConfirmButton with loading spinner", () => {
  wrapper = mount<ConfirmButtonProps>(
    <ConfirmButton {...props} loading={true} />
  );
  expect(wrapper.find("Loader").exists()).toBeTruthy();
});
