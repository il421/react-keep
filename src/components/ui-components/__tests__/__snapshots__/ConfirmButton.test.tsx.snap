// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmButton 1`] = `
"<ConfirmButton text=\\"text\\" onClick={[Function: mockConstructor]}>
  <FlexBox className={[undefined]} justifyContent=\\"center\\">
    <div style={{...}} className={[undefined]}>
      <button id={[undefined]} className=\\"login-button\\" type=\\"submit\\" disabled={false} onClick={[Function: mockConstructor]}>
        text
      </button>
    </div>
  </FlexBox>
</ConfirmButton>"
`;
